stress/shell/src/main/java/alluxio/stress/cli/StressJobServiceBench.java

```diff
@@ -0,0 +1,280 @@
+import alluxio.util.FormatUtils;
+import alluxio.util.executor.ExecutorServiceFactories;
+import alluxio.worker.job.JobMasterClientContext;
+
+import com.beust.jcommander.ParametersDelegate;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.HdrHistogram.Histogram;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Job Service stress bench.
+ */
+public class StressJobServiceBench extends Benchmark<JobServiceBenchTaskResult> {
+  private static final Logger LOG = LoggerFactory.getLogger(StressJobServiceBench.class);
+  public static final int MAX_RESPONSE_TIME_BUCKET_INDEX = 0;
+  @ParametersDelegate
+  private JobServiceBenchParameters mParameters = new JobServiceBenchParameters();
+
+  /**
+   * Creates instance.
+   */
+  public StressJobServiceBench() {}
+
+  /**
+   * @param args command-line arguments
+   */
+  public static void main(String[] args) {
+    mainInternal(args, new StressJobServiceBench());
+  }
+
+  @Override
+  public void prepare() throws Exception {
+    FileSystemContext fsContext =
+        FileSystemContext.create(new InstancedConfiguration(ConfigurationUtils.defaults()));
+    FileSystem fileSystem = FileSystem.Factory.create(fsContext);
+    long start = CommonUtils.getCurrentMs();
+    deleteBasePath(fileSystem);
+    long deleteEnd = CommonUtils.getCurrentMs();
+    LOG.info("Cleanup delete took: {} s", (deleteEnd - start) / 1000.0);
+    createFiles(fileSystem, mParameters.mNumFilesPerDir, mParameters.mNumDirs,
+        mParameters.mFileSize);
+    long createEnd = CommonUtils.getCurrentMs();
+    LOG.info("Create files took: {} s", (createEnd - deleteEnd) / 1000.0);
+  }
+
+  private void createFiles(FileSystem fs, int numFiles, int numDirs, int fileSize)
+      throws IOException, AlluxioException {
+    CreateFilePOptions options =
+        CreateFilePOptions.newBuilder().setRecursive(true).setWriteType(WritePType.THROUGH).build();
+    for (int dirId = 0; dirId < numDirs; dirId++) {
+      for (int fileId = 0; fileId < numFiles; fileId++) {
+        String filePath =
+            String.format("%s/%s/%d/%d", mParameters.mBasePath, mBaseParameters.mId, dirId, fileId);
+        createByteFile(fs, new AlluxioURI(filePath), options, fileSize);
+      }
+    }
+  }
+
+  private void createByteFile(FileSystem fs, AlluxioURI fileURI, CreateFilePOptions options,
+      int len) throws IOException, AlluxioException {
+    try (FileOutStream os = fs.createFile(fileURI, options)) {
+      byte[] arr = new byte[len];
+      for (int k = 0; k < len; k++) {
+        arr[k] = (byte) k;
+      }
+      os.write(arr);
+    }
+  }
+
+  private void deleteBasePath(FileSystem fs) throws IOException, AlluxioException {
```

Comment:
similarly this delete is going to happen n times, once for every worker?

Reply:
yeah, you are right, this would cause delete many times even there's a check in the beginning... an idea would be let workers delete their own directory to avoid many workers delete the whole base directory. Another thought would be something mentioned by Bowen, we have a preparation stage for the whole cluster and another preparation stage for each worker(this would include larger change).

Reply:
maybe just add a TODO here? delete is less of a concern, we just have to make sure repeated deletes do not cause an error.

Reply:
good catch @ssz1997 , maybe we do need to delete individual directories then.